kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/db/H2DbHelper.java

```diff
@@ -45,109 +32,24 @@ protected void unsetConfigurationService(ConfigurationService configurationServi
         this.configurationService = null;
     }
 
-    protected void activate(Map<String, Object> properties) {
-        logger.info("activating...");
-        startDefaultDbServiceInstance();
-        updated(properties);
-        logger.info("activating...done");
-    }
-
-    protected void updated(Map<String, Object> properties) {
-        logger.info("updating...");
-        restartServer(new ServerConfiguration(properties));
-        logger.info("updating...done");
-    }
-
-    protected void deactivate() {
-        logger.info("deactivating...");
-        shutdownServer();
-        logger.info("deactivating...done");
-    }
-
     private void startDefaultDbServiceInstance() {
         try {
             if (configurationService.getComponentConfiguration(H2DbService.DEFAULT_INSTANCE_PID) == null) {
+                logger.info("Default H2DbService instance configuration not found, creating new instance...");
                 configurationService.createFactoryConfiguration(H2_DB_SERVICE_FACTORY_PID,
                         H2DbService.DEFAULT_INSTANCE_PID, null, true);
+            } else {
+                logger.info("Default H2DbService instance configuration found");
             }
         } catch (KuraException e) {
             throw new ComponentException(e);
```

Comment:
As far as I remember this causes the framework to retry component activation. Did you choose this for a specific reason?